### PR TITLE
Create models

### PR DESCRIPTION
**Pull request by vecerap**
I created various types of models. I took inspiration from wireframe.

## Intended usecases for models
- Detail models are used for input forms, where the validation is present
- List models are used for list pages eg. On user page showing quizzes
- Result models are on result pages after the user completes the quiz, it contains comparison properties which will be joined from binding entites.

I want to open a discussion here about the usecases of the models and whether they contain all the properties which will be required to display or send via API

### CODE REVIEW

File: src/Quizzer.API.BL/Models/AnswerModel.cs
Comment (KoubaFromPucks) [inline src/Quizzer.API.BL/Models/AnswerModel.cs:1]: Would´t it be better if the name was `AnswerDetailModel`?

File: src/Quizzer.API.BL/Models/QuestionModel.cs
Comment (KoubaFromPucks) [inline src/Quizzer.API.BL/Models/QuestionModel.cs:1]: As above i am missing word `Detail` in class name.

File: src/Quizzer.API.BL/Models/QuizDetailModel.cs
Comment (KoubaFromPucks) [inline src/Quizzer.API.BL/Models/QuizDetailModel.cs:1]: So it will not be possible to change `Name`, `Description` and add or remove Questions?

Comment (vecerap) [reply src/Quizzer.API.BL/Models/QuizDetailModel.cs:1]: Sorry, missed that, Intellisense suggestions...

File: src/Quizzer.API.BL/Models/AnswerModel.cs
Comment (vecerap) [reply src/Quizzer.API.BL/Models/AnswerModel.cs:1]: Fixed

File: src/Quizzer.API.BL/Models/QuestionModel.cs
Comment (vecerap) [reply src/Quizzer.API.BL/Models/QuestionModel.cs:1]: Fixed

**Review by KoubaFromPucks — APPROVED** [review]

**Review by brtkkndrl — APPROVED** [review]

**Pull request merged**